src/snowflake/snowpark/table.py

```diff
@@ -664,6 +686,10 @@ def drop_table(self) -> None:
         Note that subsequent operations such as :meth:`DataFrame.select`, :meth:`DataFrame.collect` on this ``Table`` instance and the derived DataFrame will raise errors because the underlying
         table in the Snowflake database no longer exists.
         """
-        self._session.sql(
-            f"drop table {self.table_name}"
-        )._internal_collect_with_tag_no_telemetry()
+        if hasattr(self._session._conn, "entity_registry"):
```

Comment:
This is interesting. Should we do `isinstance` check instead?

Reply:
updated to use the isinstance check to be consistent across the code base